src/core/http/response.rs

```diff
@@ -18,6 +18,29 @@ pub struct Response<Body> {
     pub body: Body,
 }
 
+pub trait FromValue {
+    fn from_value(value: serde_json_borrow::Value) -> Self;
+}
+
+impl FromValue for ConstValue {
+    fn from_value(value: serde_json_borrow::Value) -> Self {
+        match value {
+            serde_json_borrow::Value::Null => ConstValue::Null,
+            serde_json_borrow::Value::Bool(b) => ConstValue::Boolean(b),
+            serde_json_borrow::Value::Number(n) => ConstValue::Number(n.into()),
+            serde_json_borrow::Value::Str(s) => ConstValue::String(s.into()),
+            serde_json_borrow::Value::Array(a) => {
+                ConstValue::List(a.into_iter().map(|v| Self::from_value(v)).collect())
+            }
+            serde_json_borrow::Value::Object(o) => ConstValue::Object(
+                o.into_iter()
+                    .map(|(k, v)| (Name::new(k), Self::from_value(v)))
+                    .collect(),
+            ),
+        }
+    }
+}
```

Comment:
Add significant documentation about why are we doing deserialization this way. Helpful for future developers, who don't delete this optimization as a part of some refactor.